### PR TITLE
View Childcare Dashboard

### PR DESCRIPTION
**Pull request by silbermm**
Create childcare dashboard at `/childcare-dashboard` that only allows a head of household to view.

### CODE REVIEW

File: Gateway/crds-angular/Services/ChildcareService.cs [hunk 1]
```diff
@@ -228,6 +235,85 @@ public void RejectChildcareRequest(int childcareRequestId, ChildcareRequestDto c
...
+                                EventDate = eventDetails.EventStartDate.Date,
+                                Cancelled = eventDetails.Cancelled
+                            });
+                        }
+
+                        //Date exists, add group
+                        var eventGroup = _eventService.GetEventGroupsForEvent(eventDetails.EventId, token).FirstOrDefault(g => g.GroupTypeId == _childcareGroupType);
+                        var ccEventGroup = _groupService.GetGroupDetails(eventGroup.GroupId);
+                        var eligibleChildren = new List<ChildcareRsvp>();
+                        foreach (var member in household)
```
Comment (silbermm) [inline Gateway/crds-angular/Services/ChildcareService.cs]: :mask:

File: Gateway/crds-angular/Services/ChildcareService.cs [hunk 2]
```diff
@@ -228,6 +235,85 @@ public void RejectChildcareRequest(int childcareRequestId, ChildcareRequestDto c
             }
         }
 
+        public ChildcareDashboardDto GetChildcareDashboard(int contactId)
```
Comment (silbermm) [inline Gateway/crds-angular/Services/ChildcareService.cs]: Is this method doing too big? How can we split it up to make smaller, more testable methods?

File: CI/SQL/20160707_090200_US3378-Add Cancelled_to_EventWithDetails_Pageview.sql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+USE MinistryPlatform
+
```
Comment (charlie-retzler) [inline CI/SQL/20160707_090200_US3378-Add Cancelled_to_EventWithDetails_Pageview.sql]: Missing GO

File: Gateway/MinistryPlatform.Translation/Repositories/GroupRepository..cs [hunk 4]
```diff
@@ -486,7 +493,7 @@ public List<MpGroupParticipant> getEventParticipantsForGroup(int groupId, int ev
 
         public List<MpGroup> GetGroupsByTypeForParticipant(string token, int participantId, int groupTypeId)
         {
-            var groupDetails = ministryPlatformService.GetPageViewRecords(MyCurrentGroupsPageView, token, String.Format(",,{0}", groupTypeId));
+            var groupDetails = ministryPlatformService.GetPageViewRecords(CurrentGroupParticipantsByGroupTypePageView, token, String.Format(",{0},,,{1}", participantId, groupTypeId));
```
Comment (silbermm) [inline Gateway/MinistryPlatform.Translation/Repositories/GroupRepository..cs]: I think the individual fields need to be quoted

File: Gateway/MinistryPlatform.Translation/Repositories/EventRepository.cs [hunk 5]
```diff
@@ -357,6 +358,27 @@ public List<MpEventGroup> GetEventGroupsForEvent(int eventId, string token)
...
+                GroupTypeId = record.ToInt("Group_Type_ID")
+            }).ToList();
+        }
+
+        public List<MpEventGroup> GetEventGroupsForGroup(int groupId, string token)
+        {
+            var searchString = string.Format(",,,\"{0}\"", groupId);
+            var records = _ministryPlatformService.GetPageViewRecords(_eventGroupsPageViewId, token, searchString);
+
+            if (records != null)
```
Comment (silbermm) [inline Gateway/MinistryPlatform.Translation/Repositories/EventRepository.cs]: I think the comparison / returning of null is checking the opposite of what it should.

File: Gateway/crds-angular/Models/Crossroads/Childcare/ChildcareDashboardDto.cs [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        public List<ChildCareDate> AvailableChildcareDates { get; set; }
+
+        public ChildcareDashboardDto()
+        {
+            AvailableChildcareDates = new List<ChildCareDate>();
+        }
+    }
+
+
+    public class ChildCareDate
```
Comment (silbermm) [inline Gateway/crds-angular/Models/Crossroads/Childcare/ChildcareDashboardDto.cs]: Can these be moved to separate files?

File: Gateway/crds-angular/Models/Crossroads/Childcare/ChildcareDashboardDto.cs [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        [JsonProperty(PropertyName = "communityGroups")]
+        public List<ChildcareGroup> Groups { get; set;}
+
+        public ChildCareDate()
+        {
+            Groups = new List<ChildcareGroup>();
+        }
+    }
+
+    public class ChildcareGroup
```
Comment (silbermm) [inline Gateway/crds-angular/Models/Crossroads/Childcare/ChildcareDashboardDto.cs]: Can these be moved to separate files?

File: Gateway/crds-angular/Models/Crossroads/Childcare/ChildcareDashboardDto.cs [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+        [JsonProperty(PropertyName = "eligibleChildren")]
+        public List<ChildcareRsvp> EligibleChildren { get; set; }
+
+        public ChildcareGroup()
+        {
+            EligibleChildren = new List<ChildcareRsvp>();
+        }
+    }
+
+    public class ChildcareRsvp
```
Comment (silbermm) [inline Gateway/crds-angular/Models/Crossroads/Childcare/ChildcareDashboardDto.cs]: Can these be moved to separate files?

File: Gateway/crds-angular/Services/ChildcareService.cs [hunk 9]
```diff
@@ -228,6 +235,85 @@ public void RejectChildcareRequest(int childcareRequestId, ChildcareRequestDto c
...
+        {
+            var dashboard = new ChildcareDashboardDto();
+            var token = _apiUserService.GetToken();
+
+            //Figure out who is a head in my household
+            var contact = _contactService.GetContactById(contactId);
+            var household = _contactService.GetHouseholdFamilyMembers(contact.Household_ID);
+            var houseHeads = household.Where(h => h.HouseholdPosition != null && h.HouseholdPosition.ToUpper().StartsWith("HEAD")); //TODO: Get rid of magic string. Household Position
```
Comment (silbermm) [inline Gateway/crds-angular/Services/ChildcareService.cs]: Is there a better way to determine head of household?

File: crossroads.net/app/childcare_dashboard/childcareDashboard.service.js [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+class ChildcareDashboardService { 
+  constructor($resource, $cookies) {
+    this.resource = $resource;
+    this.cookies = $cookies;
+    this.headOfHouseholdError = false;
+    this.dashboard = $resource(__API_ENDPOINT__ + `api/childcare/dashboard/:contactId`);
+  }
+
+  fetchChildcareDates() {
+    const contactId = this.cookies.get('userId');
```
Comment (silbermm) [inline crossroads.net/app/childcare_dashboard/childcareDashboard.service.js]: should we be getting this directly from the cookie? Are there other places that do this differently?

File: crossroads.net/app/childcare_dashboard/childcare_group/childcareGroup.html [hunk 11]
```diff
@@ -0,0 +1,42 @@
+<div class="panel panel-default childcare-dashboard">
+  <div class="panel-heading">
+    <h3 class="panel-title">
+      {{childcareGroup.communityGroup.groupName}} •
```
Comment (silbermm) [inline crossroads.net/app/childcare_dashboard/childcare_group/childcareGroup.html]: Is the dot at the end of the line supposed to be there?

File: crossroads.net/app/childcare_dashboard/childcare_group/childcareGroup.html [hunk 12]
```diff
@@ -0,0 +1,42 @@
+<div class="panel panel-default childcare-dashboard">
+  <div class="panel-heading">
+    <h3 class="panel-title">
+      {{childcareGroup.communityGroup.groupName}} • 
+      {{childcareGroup.communityGroup.groupMemberName}}</h3>
+    <span> 
+      {{childcareGroup.communityGroup.eventStartTime | date: 'h:mm'}}-{{childcareGroup.communityGroup.eventEndTime | date: 'shortTime'}}
```
Comment (silbermm) [inline crossroads.net/app/childcare_dashboard/childcare_group/childcareGroup.html]: You are using 2 different formats for date here. Not sure if it's a problem or not.

File: crossroads.net/app/routes.js [hunk 13]
```diff
@@ -419,6 +419,23 @@
             }
           }
         })
+        /*.state('childcare-dashboard', {*/
```
Comment (silbermm) [inline crossroads.net/app/routes.js]: Should the commented code be removed?

File: Gateway/crds-angular/Services/ChildcareService.cs (hunk 9 above)
Comment (silbermm) [inline Gateway/crds-angular/Services/ChildcareService.cs]: we talked about this too... we think this is the best way

**Pull request merged**